iroha/src/kura.rs

```diff
@@ -155,31 +182,35 @@ impl BlockStore {
             .await
             .wrap_err("Failed to open storage file.")?;
         let hash = block.hash();
-        let serialized_block: Vec<u8> = block.into();
+        let block: VersionedValidBlock = block.clone().try_into()?;
+        let serialized_block: Vec<u8> = block.encode_versioned()?;
         file.write_all(&serialized_block)
             .await
             .wrap_err("Failed to write to storage file.")?;
         Ok(hash)
     }
 
-    async fn read(&self, height: u64) -> Result<ValidBlock> {
+    async fn read(&self, height: u64) -> Result<VersionedValidBlock> {
         let path = self.get_block_path(height);
         let mut file = File::open(&path).await.wrap_err("No file found.")?;
         let metadata = metadata(&path).await.wrap_err("Unable to read metadata.")?;
         let mut buffer = vec![0; metadata.len() as usize];
         let _ = file.read(&mut buffer).await.wrap_err("Buffer overflow.")?;
-        Ok(ValidBlock::try_from(buffer).expect("Failed to read block from store."))
+        VersionedValidBlock::decode_versioned(&buffer).wrap_err("Failed to read block from store.")
     }
 
     /// Returns a sorted vector of blocks starting from 0 height to the top block.
-    async fn read_all(&self) -> Vec<ValidBlock> {
+    async fn read_all(&self) -> Result<Vec<ValidBlock>> {
```

Comment:
Seems like `read` and `read_all` should both return one of these block's types: `VersionedValidBlock` or `ValidBlock`